### PR TITLE
error_logger

### PR DESCRIPTION
**Pull request by mitchellhenke**

### CODE REVIEW

File: lib/timber/events/error_event.ex [hunk 1]
```diff
@@ -137,6 +155,26 @@ defmodule Timber.Events.ErrorEvent do
...
+  defp stacktrace_to_backtrace(stacktrace) do
+    # arity is an integer or list of arguments
+    Enum.map(stacktrace, fn({module, function, arity, location}) ->
+      arity = case arity do
+        arity when is_list(arity) -> length(arity)
+        _ -> arity
+      end
+
+      file = Keyword.get(location, :file)
+             |> Kernel.to_string()
```
Comment (DavidAntaramian) [inline lib/timber/events/error_event.ex]: Just for clarity, is this because the result of `Keyword.get(location, :file)` would be an IO list?

Comment (mitchellhenke) [reply lib/timber/events/error_event.ex]: yep

File: lib/timber/integrations/error_logger.ex [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+  active will cause duplicate errors to be reported to the log, which is why
+  we recommend disabling it using the following configuration option:
+
+  ```
+  config :logger, :handle_otp_reports, false
+  ```
+
+  However, the OTP report handler handles additional report types as well. If you
+  find that you would like these reports to be logged, just be aware that every
+  exception will be displayed twice in the log.
```
Comment (DavidAntaramian) [inline lib/timber/integrations/error_logger.ex]: I think you should add a note here saying that, with the default handler the report will also be broken out across multiple lines; whereas the Timber handler consolidates the information into a single line

File: lib/timber/integrations/error_logger.ex [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+
+  @behaviour :gen_event
+
+  alias Timber.Events.ErrorEvent
+
+  @doc false
+  def init(opts \\ []) do
+    max_message_count = Keyword.get(opts, :max_message_count, 300)
+    state = %{
+      logger: Process.whereis(Logger),
```
Comment (DavidAntaramian) [inline lib/timber/integrations/error_logger.ex]: For the sake of future maintainers, could you add a note before all the function declarations about what each state field is used for and how the overall architecture of this works?

File: lib/timber/integrations/error_logger.ex [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+  end
+
+  def handle_event(event , state) do
+    state = check_threshold(state)
+    log_event(event, state)
+
+    {:ok, state}
+  end
+
+  def get_metadata({:crash_report, [error_report | _neighbors]}, :error) do
```
Comment (DavidAntaramian) [inline lib/timber/integrations/error_logger.ex]: I thought we were ignoring crash reports altogether, so I'm confused as to what this code block is responsible for.

**Comment by DavidAntaramian** [comment]
Oh, and please add an entry to CHANGELOG.md

**Comment by mitchellhenke** [comment]
@DavidAntaramian Addressed your comments, let me know if it isn't clear enough.

Wasn't sure if I should squash the commit before you get a chance to review.

**Comment by binarylogic** [comment]
Yes, please squash.

**Comment by mitchellhenke** [comment]
done!

**Comment by binarylogic** [comment]
I'll let @DavidAntaramian approve and merge

**Pull request merged**